src/logging.cpp

```diff
@@ -195,10 +195,6 @@ bool GetLogCategory(BCLog::LogFlags& flag, const std::string& str)
         flag = BCLog::ALL;
         return true;
     }
-    if (str == "0") {
```

Comment:
In commit "rpc: remove misleading text from logging help" (e78866a10973c64a59e8a71eed737328b6bc0ca3)

Right now this returns NONE/true when the empty string "" is passed. Should return false in that case too and never return NONE?

Also, this is a nice change, but I found description confusing because the whole thing sounds it this like a help text change, but then the last sentence mentions it also adds a new error. Would suggest a description more like:

```
rpc: make logging method reject "0" category

Current logging RPC method documentation claims to accept "0" and "none"
categories, but the "none" argument is actually rejected and the "0" argument is
ignored. Update the implementation to refuse both categories, and remove the
help text claiming to support them.
```

Reply:
Reworded the commit message as suggested, but I didn't get this:

> Right now this returns NONE/true when the empty string "" is passed.

Hmm, no? It returns ALL/true when "" is passed, in `master` and in this PR - there is an `if (str.empty() ...` a few lines above.

> Should return false in that case too

You mean to return false when empty string is passed? That would be some further change that maybe makes sense, but is not included in this PR.

> and never return NONE?

It never returns NONE (in this PR).

Reply:
> > Right now this returns NONE/true when the empty string "" is passed.
> 
> Hmm, no? It returns ALL/true when "" is passed, in `master` and in this PR - there is an `if (str.empty() ...` a few lines above.

Right, I think I was confused when I wrote that comment. Makes sense to keep "" behavior unchanged, too.